lib/Service/AIIntergrationsService.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Hamza Mahjoubi <hamzamahjoubi22@proton.me>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use OCA\Mail\Exception\ServiceException;
+use OCP\TextProcessing\IManager;
+use OCP\TextProcessing\SummaryTaskType;
+use OCP\TextProcessing\Task;
+use function array_map;
+
+class AIIntergrationsService {
+	/** @var IManager */
+	private $manager;
+
+
+	public function __construct(IManager $manager) {
+		$this->manager = $manager;
+	}
+	/**
+	 *
+	 * @throws ServiceException
+	 */
+	public function summarizeThread(string $threadId, array $messages, $currentUserId): string {
+		if(in_array(SummaryTaskType::class, $this->manager->getAvailableTaskTypes())) {
```

Comment:
```suggestion
		if(in_array(SummaryTaskType::class, $this->manager->getAvailableTaskTypes(), true)) {
```

won't change a lot but let's use strict in_array consistently